custom/icds/views/custom_sms_report.py

```diff
@@ -83,19 +86,14 @@ def set_error_messages(self, reports, start_date, end_date):
 
 
 def _prepare_display_message(reports_in_progress, report_count):
-    message = _('Reports in progress: ').format(
-        reports_count=report_count,
-    )
+    message = _('Reports in progress: ')
     for index, report in enumerate(reports_in_progress):
-        message += '"{start_date} -- {end_date}"'.format(
-            start_date=report['start_date'],
-            end_date=report['end_date']
-        )
+        message += '"{report_id}"'.format(report_id=report)
         if index != report_count - 1:
             message += ', '
     return message
 
 
 def _report_already_in_progress(reports, start_date: str, end_date: str):
-    return any(report['start_date'] == str(start_date) and report['end_date'] == str(end_date)
-        for report in reports)
+    report_id = start_date + '--' + end_date
```

Comment:
this logic to generate report_id out of start data and end date should be at one place. If someday we need to change it, this will get very difficult.